### PR TITLE
adds configurable dbt cloud host

### PR DESCRIPTION
**Pull request by ernestoongaro**
## Description & motivation
This is to satisfy https://github.com/data-mie/dbt-cloud-cli/issues/38

## Checklist
~~[ ] I have written unit tests for new features (e.g., a new command test case in [conftest.py](../tests/conftest.py))~~
~~[ ] I have written integration tests for new features (e.g., a new step in the [CI workflow](workflows/ci.yml))~~
- [x] I have added descriptions to new commands and arguments
- [x] I have updated the README.md (if applicable)

### CODE REVIEW

**Comment by ernestoongaro** [comment]
@stumelius I'm going to need a lot of help with the unit and integration tests. Unfortunately for testing urls other than cloud.getdbt.com I don't have a great answer for how we will test them, but effectively the idea is that the functionality is the same, just a different URL.

**Review by stumelius — CHANGES_REQUESTED** [review]

**Comment by stumelius** [comment]
Thanks for the PR @ernestoongaro ❤️ I left you a small request for change in the review. There's no need to write new unit or integration tests for this since I have no single tenant instance to test against. As soon as the existing tests pass this PR is good for merging :)

File: dbt_cloud/field.py [hunk 1]
```diff
@@ -1,6 +1,11 @@
 import os
 from pydantic import Field
 
+DBT_CLOUD_HOST_FIELD = Field(
+    default_factory=lambda: 'cloud.getdbt.com',
```
[the remark this answers is not in the document]
Comment (stumelius) [reply dbt_cloud/field.py:5]: Can you also document the `DBT_CLOUD_HOST` env var in README? :)

[the remark this answers is not in the document]
Comment (ernestoongaro) [reply dbt_cloud/field.py]: Great! I have added to the readme and set the default as you suggested- great idea!

**Comment by ernestoongaro** [comment]
@stumelius Regarding `black` - I'm getting:

```(dbt_cloud_api_dev) ernesto@Ernesto-Ongaro dbt-cloud-cli % black --check .
All done! ✨ 🍰 ✨
25 files would be left unchanged.```

I don't think that the CI has run again with my changes though?

**Comment by stumelius** [comment]
Looks good! The integration testing job is failing because it requires secrets from this repo which are not available for forks. I manually verified that the multi-tenant commands are still working 👍

**Pull request merged**